### PR TITLE
Refactor TransferWalletBalance

### PR DESCRIPTION
**Pull request by MariaHCD**
Related Web PR: https://github.com/Expensify/Web-Expensify/pull/34127

### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/216150

### Tests

Pre-requisite accounts (via clitools.sh `generator:account` and `validator:wallet`)
1. Account A set up with a gold wallet and a validated bank account
2. Account B set up with a gold wallet and a validated bank account

Tests:
1. As Account A send money to Account B
2. As Account B, navigate to Settings > Payments > Transfer Balance

Screen Shot 2022-07-08 at 9 37 51 PM

Screen Shot 2022-07-08 at 9 37 57 PM

3. Click on `Transfer Balance` and then the `Transfer $<amount>` button

Screen Shot 2022-07-08 at 9 38 53 PM

4. Verify the transfer completes successfully and you see the success page

Screen Shot 2022-07-11 at 1 28 08 PM

5. Verify clicking `Done` or the close button directs you back to the Settings > Payments page

Screen Shot 2022-07-08 at 10 11 44 PM

6. Verify the `Transfer Balance` button on the Payments page is disabled when offline

Screen Shot 2022-07-21 at 12 14 19 PM

7. Verify that the button on the Transfer Balance page is disabled when offline

Screen Shot 2022-07-21 at 12 14 41 PM

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [ ] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I ran the tests on **all platforms** & verified they passed on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose and it is
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn’t already exist
    - [x] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps

Pre-requisite accounts:
1. Account A on the wallet beta, set up with a gold wallet and a validated bank account
2. Account B on the wallet beta, set up with a gold wallet and a validated bank account

Tests:
1. As Account A send money to Account B
2. As Account B, navigate to Settings > Payments > Transfer Balance
3. Click on `Transfer Balance` and then the `Transfer $<amount>` button
4. Verify the transfer completes successfully and you see the success page
5. Verify clicking `Done` or the close button directs you back to the Settings > Payments page
6. Verify the `Transfer Balance` button on the Payments page is disabled when offline

Screen Shot 2022-07-21 at 12 14 19 PM

7. Verify that the button on the Transfer Balance page is disabled when offline

Screen Shot 2022-07-21 at 12 14 41 PM

- [ ] Verify that no errors appear in the JS console

#### Web

https://user-images.githubusercontent.com/12268372/178234139-06a7a36f-4c48-4fb8-b317-faad70afa512.mp4

#### Desktop

https://user-images.githubusercontent.com/12268372/178480028-3c14d78d-e063-4386-b015-1228cff91da0.mp4

### CODE REVIEW

File: src/pages/settings/Payments/TransferBalancePage.js
Comment (shawnborton) [inline src/pages/settings/Payments/TransferBalancePage.js:184]: I left this in the other issue, but here is a blue version in case you want to switch it out: [tada--blue.svg.zip](https://github.com/Expensify/App/files/9075044/tada--blue.svg.zip)

Comment (MariaHCD) [reply src/pages/settings/Payments/TransferBalancePage.js]: Thanks, Updated!

Screen Shot 2022-07-11 at 1 28 08 PM

Comment (shawnborton) [reply src/pages/settings/Payments/TransferBalancePage.js]: Looks good!

**Review by thienlnam — CHANGES_REQUESTED** [review]
Looking good! Just a few comments

File: src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js
Comment (MariaHCD) [reply src/pages/settings/Payments/PaymentsPage/BasePaymentsPage.js:423]: Yep, removing the confirm modal is intentional, we used to show it after a successful transfer:

Screen Shot 2022-06-16 at 1 43 36 PM

But we decided to make it consistent with other success screens instead. Yeah, I think `walletTransfer.transferAmount` isn't used anywhere else. `shouldShowConfirmModal` and `dismissWalletConfirmModal` aren't used either. Updated!

**Comment by JmillsExpensify** [comment]
Shall we remove the hold now that Web-E PR is merged?

**Comment by stitesExpensify** [comment]
@JmillsExpensify we need to wait until the web PR is deployed to production to prevent this from getting merged when it won't work yet :)

File: src/libs/actions/PaymentMethods.js
Comment (thienlnam) [inline src/libs/actions/PaymentMethods.js:233]: Do we still use the transferAmount here as well?

**Review by ctkochan22 — DISMISSED** [review]

**Comment by ctkochan22** [comment]
Should be off hold now?

File: src/libs/actions/PaymentMethods.js
Comment (MariaHCD) [reply src/libs/actions/PaymentMethods.js]: Thanks for catching that! Removed.

**Comment by MariaHCD** [comment]
Updated and off hold!

**Review by ctkochan22 — DISMISSED** [review]

File: src/pages/settings/Payments/TransferBalancePage.js
Comment (stitesExpensify) [inline src/pages/settings/Payments/TransferBalancePage.js:186]: We don't want to always show this right? Transferring to a debit card is instant

**Review by stitesExpensify — CHANGES_REQUESTED** [review]
Code LGTM, just one small change!

File: src/pages/settings/Payments/TransferBalancePage.js
Comment (MariaHCD) [reply src/pages/settings/Payments/TransferBalancePage.js:186]: That's a good point, re-opening the mockup issue to see if we can get another copy in the case of debit card transfers: https://github.com/Expensify/Expensify/issues/217387

Comment (MariaHCD) [reply src/pages/settings/Payments/TransferBalancePage.js]: Just waiting on the Spanish translation and then we should be good to go!

**Review by MariaHCD — COMMENTED** [review]
Updated with the correct copy and translations for the success screen. All good for re-review!

Also, disabled the transfer buttons when offline. Although it looks like the alignment of `You appear to be offline` is a little off (which can be fixed in a follow up PR):

Screen Shot 2022-07-18 at 12 12 51 PM

Screen Shot 2022-07-18 at 12 13 08 PM

**Comment by shawnborton** [comment]
I agree that the alignment is off. I commented in [Slack here](https://expensify.slack.com/archives/C02HWMSMZEC/p1658162822563989) about that.

**Comment by MariaHCD** [comment]
Merged https://github.com/Expensify/App/pull/9959 into this branch and added the OfflineIndicator to:

 1. BasePaymentsPage:

Screen Shot 2022-07-19 at 11 31 55 AM

2. TransferBalancePage: I think the padding on the bottom is a little too much but that's set on the FixedFooter component and I'm not sure how to adjust that without affecting the styles of the footer when online.

Screen Shot 2022-07-19 at 11 39 44 AM

Let me know what you think, @shawnborton @ctkochan22

**Comment by shawnborton** [comment]
In terms of the offline indicator itself, I think we want it to be something like this:
image


Where there is 20px from the left edge to the icon, and below there should be 12px of spacing. Between the icon and the text, there should be a 12px gap. 

Not sure if that's what's going on here or if you are even making those changes here, but just wanted to call it out.

**Comment by MariaHCD** [comment]
Updated the styles for the OfflineIndicator, specifically:
- Updated the icon and altered the gap between the icon and text to be 12px
- On the Payments page and on the TransferBalance page, there is 20px from the left edge to the icon, and below there is 12px of spacing

Relevant commits: 
- https://github.com/Expensify/App/pull/9589/commits/1796df02ea1f4a914219e0466cd04cc9f0297894
- https://github.com/Expensify/App/pull/9589/commits/7909bbbe8aef4637f5c96276bfceafe99762b6a5

Screen Shot 2022-07-20 at 3 24 34 PM

Screen Shot 2022-07-20 at 4 32 57 PM

**Review by thienlnam — DISMISSED** [review]

File: src/components/OfflineIndicator.js
Comment (shawnborton) [inline src/components/OfflineIndicator.js:41]: @MariaHCD Just wanted to confirm, this icon should be shown at 16x16. I think that's the case from seeing the mockups, but just double checking!

**Review by stitesExpensify — DISMISSED** [review]

File: src/components/OfflineIndicator.js
Comment (MariaHCD) [reply src/components/OfflineIndicator.js]: It was set at 12x12 but I've updated to 16x16, thanks!

Comment (MariaHCD) [reply src/components/OfflineIndicator.js]: Screen Shot 2022-07-21 at 12 14 19 PM

Screen Shot 2022-07-21 at 12 14 41 PM

**Comment by MariaHCD** [comment]
Fixed merge conflicts and tweaked the styles for the OfflineIndicator

**Review by shawnborton — DISMISSED** [review]

**Review by thienlnam — DISMISSED** [review]
Approved but pending another merge conflic

**Review by ctkochan22 — DISMISSED** [review]
Looks good. Just conflicts

**Comment by ctkochan22** [comment]
@MariaHCD I hope its okay, I feel like we have all the approves, but cause of the timezone, we are getting approvals after changes that causes conflicts. I just went in and fixed the conflicts

**Comment by ctkochan22** [comment]
@stitesExpensify @shawnborton @thienlnam Can we get a quick re-reveiw, I just took care of the small merge conflict

**Review by thienlnam — APPROVED** [review]

**Comment by thienlnam** [comment]
Just going to merge this before we get any more conflicts - all previous reviewers have approved pending conflicts

**Comment by MariaHCD** [comment]
Thanks for moving this forward!

**Comment by mvtglobally** [comment]
@MariaHCD @JmillsExpensify @thienlnam Can you QA this internally? We are not able to test wallet due to PROD KYC wall

**Comment by MariaHCD** [comment]
> @MariaHCD @JmillsExpensify @thienlnam Can you QA this internally? We are not able to test wallet due to PROD KYC wall

Asked here! https://expensify.slack.com/archives/C02HWMSMZEC/p1658825528148109?thread_ts=1657045889.543309&cid=C02HWMSMZEC

**Comment by luacmartins** [comment]
@MariaHCD did we manage to test this on staging?

**Comment by yuwenmemon** [comment]
Tested on staging! It's a pass 🎉

**Pull request merged**
